testing/sync-test/src/auth.rs

```diff
@@ -32,6 +32,7 @@ pub struct TestClient {
     pub device: Device,
     // XXX do this more generically...
     pub autofill_store: Arc<AutofillStore>,
+    autofill_db_name: String,
```

Comment:
As I was updating this PR, I noticed that the two autofill `TestClient` instances were referencing the same db instance. I created a test where I first checked that both clients' databases were empty, I added a record to the first client, and then could assert that it existed on both. 🥴 

This is my attempt at resolving this issue, but I'm open to suggestions if this isn't ideal.